### PR TITLE
Change masks plugin

### PR DESCRIPTION
**Pull request by Gataniel**

### CODE REVIEW

File: app/assets/javascripts/transactions.js [hunk 1]
```diff
@@ -99,3 +109,9 @@ function datepicker_init(selector) {
     autoclose: true
   });
 }
+
+function add_masks() {
+  $('.amount').inputmask('customized_currency');
```
Comment (shir) [inline app/assets/javascripts/transactions.js]: @Gataniel
```
function addTransactionFormMasks() {
  $("form.transaction input[name='transfer[amount]']").inputmask('customized_currency')
}
```

File: app/assets/javascripts/transactions.js [hunk 2]
```diff
@@ -93,9 +97,39 @@ function show_hide_period_additional_input() {
   }
 }
 
-function datepicker_init(selector) {
+function addRemoveHints(fromCurr, toCurr) {
+  if (fromCurr != undefined && toCurr != undefined && fromCurr != toCurr ) {
```
Comment (AlexanderUlitin) [inline app/assets/javascripts/transactions.js]: @Gataniel let's write like this:

```
if (fromCurr == undefined || toCurr == undefined || fromCurr = toCurr) {
  return;
}
```

File: app/assets/javascripts/transactions.js [hunk 3]
```diff
@@ -93,9 +97,39 @@ function show_hide_period_additional_input() {
...
+      $('.transfer_exchange_rate').append('<span class="help-block">' + rate_hint + '</span>');
+    }
+
+    var amount = parseFloat($('#transfer_amount').val().replace(/\,/g,''));
+    var rate = parseFloat($('#transfer_exchange_rate').val().replace(/\,/g,''));
+    if (amount && rate) {
+      var end_sum = (amount * rate).toFixed(2);
+      if ($('.transfer_reference_id .help-block').html() != end_sum) {
+        $('.transfer_reference_id .help-block').remove();
+        $('.transfer_reference_id').append('<span class="help-block">' + end_sum + '</span>');
```
Comment (AlexanderUlitin) [inline app/assets/javascripts/transactions.js]: @Gataniel move `$('.transfer_reference_id')` to variable like `$transferReference = $('.transfer_reference_id');`

File: app/assets/javascripts/transactions.js [hunk 4]
```diff
@@ -51,39 +44,50 @@ $(function () {
     $('#new_transaction').show();
   });
 
-  $(document).on('change', '#transfer_bank_account_id', function(e) {
-    if ($('#transfer_exchange_rate').size()) {
-      show_hide_exchange_rate();
-    }
+  $(document).on('change', '#transfer_amount', function(e) {
```
Comment (AlexanderUlitin) [inline app/assets/javascripts/transactions.js]: @Gataniel combile it to `$(document).on('change', '#transfer_amount, #transfer_exchange_rate', function(e) {`

File: app/assets/javascripts/transactions.js [hunk 5]
```diff
@@ -93,9 +97,39 @@ function show_hide_period_additional_input() {
...
     autoclose: true
   });
 }
+
+function addTransactionFormMasks() {
+  $("form.transaction input[name='transaction[amount]']").inputmask('customized_currency');
+}
+
+function addTranferFormMasks() {
+  $("form.transfer input[name='transfer[amount]']").inputmask('customized_currency');
```
Comment (AlexanderUlitin) [inline app/assets/javascripts/transactions.js]: @Gataniel move to variable `var $form = $("form.transfer");`

File: app/models/bank_account.rb [hunk 6]
```diff
@@ -64,4 +62,14 @@ def set_initial_residue
...
     "#{name.truncate(30)} (#{Money::Currency.new(currency).symbol})"
   end
+
+  class << self
+    def grouped_by_currency(def_currency)
+      currencies = Currency.ordered(def_currency)
+
+      all.group_by(&:currency).sort_by do |ba|
+        currencies.index(ba.first)
+      end
```
Comment (AlexanderUlitin) [inline app/models/bank_account.rb]: @Gataniel ask Ildar how to refactor it, please.

**Pull request merged**